### PR TITLE
2016 01 cc captcha ui+1

### PR DESCRIPTION
**Pull request by local-girl**
Behold! Captcha UI, Don't forget you need to update these lines:

adhocracy.thentos_captcha.frontend_url = https://a3-stage.liqd.net/thentos/frontend/
adhocracy.thentos_captcha.backend_url = https://a3-stage.liqd.net/thentos/backend/

in
etc/frontend_development.ini.in
etc/frontend_development.ini
etc/development.ini.in
etc/development.ini

and restart your server to see captcha.

Icons need to be added to all other sites too but I want this to be approved before I do that

@fisx you might want to see @xi check UI please ?

### CODE REVIEW

**Comment by MagdaN** [comment]
I tested it - it looks nice but I get an backend error: ValueError: Expecting value: line 1 column 1 (char 0); time: 2016-01-20 19:10:24,975

@fisx :  can you also check

File: src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "TR__ACTIVATION_ERROR_SUPPORT": "Um einen neuen Link zu erhalten, kontaktiere bitte unser Support-Team und gib dabei die E-Mail-Adresse an, mit der Du Dich registrierst hast.",
     "TR__ACTIVATION_ERROR_TITLE": "Hoppla, das ging schief!",
     "TR__ACTIVATION_SUCCESS_TITLE": "Dein Account ist nun aktiv!",
+    "TR__TOGGLE_AUDIO": "Toggle Audio",
```
Comment (fisx) [inline src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json]: *toggle* is not a German word.  (-:

*Umschalten zwischen Audio und Bild"?

File: src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json [hunk 2]
```diff
@@ -33,6 +34,7 @@
     "TR__EDIT_PROFILE_IMAGE": "Profilbild bearbeiten",
     "TR__EMAIL": "E-Mail",
     "TR__EMAIL_OR_USERNAME": "E-Mail oder Benutzername",
+    "TR__ENTER_CAPTCHA": "Captcha eingeben",
```
Comment (fisx) [inline src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json]: "Captcha lösen"?  "Lösung eingeben"?  If this is the error message displayed when the input field is empty, then "Captcha eingeben" is also ok.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts [hunk 3]
```diff
@@ -361,12 +362,20 @@ export var registerDirective = (
...
+                    }
+                },
+                toggleAudio: () => {
+                    if (scope.captcha.audioEnabled === false) {
+                        scope.captcha.audioEnabled = true;
                     } else {
-                        fetchCaptchaImage(adhConfig, $sce, $http, scope);
+                        scope.captcha.audioEnabled = false;
                     }
+                    scope.captcha.refreshCaptcha();
```
Comment (fisx) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/User/Views.ts]: this looks much nicer now!

**Comment by fisx** [comment]
@MagdaN: I couldn't reproduce your backend error.  Is that in the browser console?  In a logfile?  (Which?)

I pushed a small change (code looks much cleaner now, and I could remove a few more lines due to @local-girl's refactorings).

LGTM!  (no opinion on the css code, as always)

File: src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json (hunk 2 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json]: It's not the error message it's just the label for the input

File: src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json (hunk 1 above)
Comment (local-girl) [inline src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json]: OOps fixed "Audio umschalten"

Comment (fisx) [inline src/adhocracy_frontend/adhocracy_frontend/static/i18n/core_de.json]: > OOps fixed "Audio umschalten"

That's switching from something to something other inside the context
of audio.  "Auf Audio umschalten" would work, but then the text would
have to change to "Auf Bild umschalten" when in audio mode.

Tricky and tedious, this language.  We can also leave it as is I
guess.

**Pull request merged**